임현순/Week1/boj13913/boj13913.cpp

```diff
@@ -0,0 +1,46 @@
+#include <iostream>
+#include <queue>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+int visited[150004];
+int before_pos[150004];
+int n, k, temp;
+queue<int> q;
+vector<int> v;
+
+int main()
+{
+    cin >> n >> k;
+    visited[n] = 1;
+    q.push(n);
+    while (q.size())
+    {
+        temp = q.front();
+        q.pop();
+        if (temp == k)
+            break;
+        for (int i : {temp + 1, temp - 1, temp * 2})
+        {
+            if (i < 0 || i > 150000 || visited[i] != 0)
+                continue;
+            visited[i] = visited[temp] + 1;
+            before_pos[i] = temp;
+            q.push(i);
+        }
+    }
+
+    cout << visited[temp] - 1 << '\n';
+    while (temp != n)
+    {
+        v.push_back(temp);
+        temp = before_pos[temp];
+    }
+    v.push_back(n);
+    reverse(v.begin(), v.end());
```

Comment:
reverse를 이용하여 앞에서부터 출력하는 방법도 좋군요! 
개인적으로 c++의 장점인 stl의 reverse_iterator를 사용하는 방법도 괜찮을 것 같습니다.
```suggestion
    for (auto i = v.rbegin(); i != v.rend(); ++i) 
```

Reply:
C++이 익숙치 않아서 이런 방법이 있는줄 몰랐네요! 하나 배웠습니다 감사합니다 :)